### PR TITLE
[mono][interp] Fix arg alignment issues when mixing with compiled code

### PR DESCRIPTION
**Pull request by BrzVlad**

### CODE REVIEW

**Comment by kg** [comment]
I don't know how to push to your PR, but try https://github.com/dotnet/runtime/commit/ee960937048b9417191a08f778e541986e1abb53 to update the jiterpreter and turn it all back on. I did some basic testing in AOT mode (the affected part, I think) and it seems to work now. You'll want to revert the changes to the threshold + stats options in options-def before merging (I lowered the threshold to make sure we actually exercise it during test runs)

File: src/mono/mono/utils/options-def.h [hunk 1]
```diff
@@ -95,7 +95,7 @@ DEFINE_BOOL(jiterpreter_call_resume_enabled, "jiterpreter-call-resume-enabled",
 //  stats for options like estimateHeat, but raises overhead.
 DEFINE_BOOL(jiterpreter_disable_heuristic, "jiterpreter-disable-heuristic", FALSE, "Always insert trace entry points for more accurate statistics")
 // Automatically prints stats at app exit or when jiterpreter_dump_stats is called
-DEFINE_BOOL(jiterpreter_stats_enabled, "jiterpreter-stats-enabled", FALSE, "Automatically print jiterpreter statistics")
+DEFINE_BOOL(jiterpreter_stats_enabled, "jiterpreter-stats-enabled", TRUE, "Automatically print jiterpreter statistics")
```
Comment (kg) [inline src/mono/mono/utils/options-def.h:98]: The changes to options-def can be reverted now that the tests passed on CI

File: src/mono/mono/mini/interp/transform.c [hunk 2]
```diff
@@ -6889,7 +6947,7 @@ generate_code (TransformData *td, MonoMethod *method, MonoMethodHeader *header,
 				default: {
 					GString *res = g_string_new ("");
 					mono_type_get_desc (res, m_class_get_byval_arg (klass), TRUE);
-					g_print ("STELEM: %s -> %d (%s)\n", m_class_get_name (klass), mint_type (m_class_get_byval_arg (klass)), res->str);
+					g_print ("STELEM: %s -> %d (%s)\n", m_class_get_name (klass), mono_mint_type (m_class_get_byval_arg (klass)), res->str);
```
Comment (kg) [inline src/mono/mono/mini/interp/transform.c:6950]: Should this g_print be guarded by a td->verbose check or something?

Comment (BrzVlad) [reply src/mono/mono/mini/interp/transform.c:6950]: this is assertion code

**Review by kg — APPROVED** [review]
Everything looks good to me, once options-def is reverted

File: src/mono/mono/mini/interp/interp.c [hunk 3]
```diff
@@ -8198,22 +8144,20 @@ interp_run_clause_with_il_state (gpointer il_state_ptr, int clause_index, MonoOb
...
 	findex = 0;
 	if (sig->ret->type != MONO_TYPE_VOID)
 		findex ++;
 	if (sig->hasthis) {
 		// FIXME: This
-		sp_args++;
 		findex ++;
 	}
 	for (int i = 0; i < sig->param_count; ++i) {
+
```
Comment (kotlarmilos) [inline src/mono/mono/mini/interp/interp.c:8155]: Nit: remove new line

**Pull request merged**